### DISCUSSION TITLE
Why does stripImage increase file size?

### DISCUSSION THREAD

**Discussion by halaei** [ImageMagick/ImageMagick#1997] [General] [answered]
It seems strip can increase filesize. Can somebody tell why/where this might happen? Is this a bug or somehow my image is corrupted?
#### Commands
`convert -strip in.jpg out.jpg`
```php
$img = new Imagick('in.jpg');
$img->stripImage();
$img->writeImage('out.jpg');
```
### Example
Strip increase the size of the given file from 95KB to 98KB:

### filesize decrease on images with profile/comments
Strip usually decrease the size of images that has no profile/comments (according to PHP `Imagick::getImageProperties()`. A comment on how this happens is also appreciated.

**Comment by snibgo** [comment, marked as answer]
"-strip" will remove metadata, if there is any that can be removed.

But the image is decompressed and decoded, then encoded and compressed. There are no guarantees that this will take the same number of bytes.

**Comment by halaei** [comment]
@snibgo Thanks. I didn't know it decode and encode image. Will this affect image quality?
Also is it technically possible at all to remove these metadata without changing the actual graphical data?

**Comment by snibgo** [comment]
JPEG compression is lossy. This means that every time IM writes JPEG, it will change pixel values (almost certainly).

For changing metadata of JPEG files, I suggest using exiftool, not IM.
